annotations_ingester/annotations_generator.py

```diff
@@ -0,0 +1,196 @@
+import logging
+import re
+import sys
+from pathlib import Path
+from typing import Sequence
+
+import boto3
+import botocore
+import pystac
+from pulsar import Message
+from pystac import Catalog, Collection, STACTypeError
+from rdflib import DCAT, DCTERMS, RDF, Graph, Literal, URIRef
+
+from annotations_ingester.messagers import CatalogueSTACChangeMessager, Messager
+
+from rdflib import Graph
+
+
+CATALOGUE_PUBLIC_BUCKET_PREFIX = "/catalogs/supported-datasets/ceda-stac-catalogue/collections/"
+
+
+def is_file_immutable(file_contents):
```

Comment:
There's an easier way to tell than this: the annotations files are immutable (they have a unique filename with a UUID in it and the file never changes once created) and the other ones are not (they're named after the dataset so if the dataset changes so does the file).

Reply:
Updated. Assumes all files are immutable as discussed